tests/functional/recovery_key.js

```diff
@@ -153,5 +168,113 @@ registerSuite('Recovery key', {
         .then(fillOutSignIn(email, NEW_PASSWORD))
         .then(testElementExists(selectors.SETTINGS.HEADER));
     },
+
+    'can not re-use recovery key': function () {
+      const query = {showAccountRecovery: true};
+      return this.remote
+        .then(openPage(RESET_PASSWORD_URL, selectors.RESET_PASSWORD.HEADER, {
+          query,
+          webChannelResponses: {
+            'fxaccounts:fxa_status': {capabilities: null, signedInUser: null}
+          }
+        }))
+        .then(fillOutResetPassword(email))
+        .then(testElementExists(selectors.CONFIRM_RESET_PASSWORD.HEADER))
+        .then(openVerificationLinkInSameTab(email, 2))
+        .then(testElementExists(selectors.COMPLETE_RESET_PASSWORD_RECOVERY_KEY.HEADER))
+
+        .then(fillOutRecoveryKey(recoveryKey))
+
+        .then(testElementExists(selectors.COMPLETE_RESET_PASSWORD.HEADER))
+        .then(fillOutCompleteResetPassword(NEW_PASSWORD, NEW_PASSWORD))
+
+        .then(testElementExists(selectors.RESET_PASSWORD_COMPLETE.HEADER))
+        .then(testElementExists(selectors.RESET_PASSWORD_COMPLETE.SUB_HEADER))
+
+        .then(testIsBrowserNotified('fxaccounts:login'))
+
+        // Attempt to re-use reset link, shows expired link
+        .then(openVerificationLinkInSameTab(email, 2))
+        .then(testElementExists(selectors.COMPLETE_RESET_PASSWORD.EXPIRED_LINK_HEADER));
+    },
   }
 });
+
+registerSuite('Recovery key - unverified session', {
+  beforeEach: function () {
+    email = TestHelpers.createEmail('sync{id}');
+
+    return this.remote.then(createUser(email, PASSWORD, {preVerified: true}))
+      // when an account is created, the original session is verified
+      // re-login to destroy original session and created an unverified one
+      .then(openPage(SIGNIN_URL, selectors.SIGNIN.HEADER))
+      .then(fillOutSignIn(email, PASSWORD))
+
+      // Perform a slight sleep here to ensure that the checks
+      // to enable the panel have been performed.
+      .sleep(500)
```

Comment:
Is this sleep needed since `findByCssSelector` in `testElementExists` will wait up to 20 seconds for the element to appear?

Reply:
I'll remove the sleep and monitor for these tests to be flaky.